lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -107,7 +107,7 @@ public static ColumnIO getArrayElementColumn(ColumnIO columnIO)
     public static Map<List<String>, ColumnDescriptor> getDescriptors(MessageType fileSchema, MessageType requestedSchema)
     {
         Map<List<String>, ColumnDescriptor> descriptorsByPath = new HashMap<>();
-        List<PrimitiveColumnIO> columns = getColumns(fileSchema, requestedSchema);
+        List<PrimitiveColumnIO> columns = ImmutableList.copyOf(getColumns(fileSchema, requestedSchema));
```

Comment:
Maybe make the immutable copy inside of `getColumns` so that any future users of this method don't need to repeat this

Reply:
fixed thanks!